### PR TITLE
Add `forcecomponents`, generalize features

### PR DESCRIPTION
**Pull request by timholy**
This breaking change adds a 3rd argument to the functions that compute
`σ` and `ϕ`, requiring the feature name as an argument. This allows one
to tune the decay-length or amplitude based on specific atom, residue,
and/or field-type.

This also establishes defaults for different fields that attempt to be a
bit more physically realistic. These have not, however, been
investigated in great detail.

Finally, `forcecomponents` can be used to compute the net force on a
residue, broken down by field. This is useful for tuning the amplitude,
using the condition that a realistic structure is in equilibrium.

### CODE REVIEW

**Comment by timholy** [comment]
Will require release of a new version of https://github.com/tmcgrath325/GaussianMixtureAlignment.jl to pass tests. Let's wait on @tmcgrath325 for that.

File: src/forces.jl [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                mgmmj = mgmms[j]
+                if keyi ∈ (:Steric, :Hydrophobe, :Aromatic)
+                    keyj = keyi
+                    haskey(mgmmj.gmms, keyj) || continue
+                    force!(@view(force[:, colidx]), gx, mgmmj.gmms[keyj]; coef = keyi === :Steric ? -1 : 1)
+                elseif keyi ∈ (:PosIonizable, :NegIonizable)
+                    for keyj in (:PosIonizable, :NegIonizable)
+                        haskey(mgmmj.gmms, keyj) || continue
+                        force!(@view(force[:, colidx]), gx, mgmmj.gmms[keyj]; coef = (keyj == keyi ? -1 : 1))
+                    end
```
Comment (tmcgrath325) [inline src/forces.jl:41]: Is it worth considering any interactions between aromatic and cationic features here?

**Review by tmcgrath325 — APPROVED** [review]
Looks good! The newest version of GaussianMixtureAlignment.jl has been registered, and tests should be passing now.

File: src/forces.jl (hunk 1 above)
Comment (timholy) [reply src/forces.jl:41]: I would love to do that (we'd have to add support in `overlap` too). Currently I am unsure how to handle the aromatic interactions. Aromatic-aromatic can be either attractive or repulsive depending on the displacement and orientation of the aromatic rings. In my current usage, I'm setting that coefficient to 0, i.e., ignoring aromatic interactions.

But cation-pi might actually be easier, as that should (I think) be purely attractive.

**Comment by timholy** [comment]
In line behind https://github.com/tmcgrath325/GaussianMixtureAlignment.jl/pull/45

File: test/runtests.jl [hunk 2]
```diff
@@ -217,6 +220,79 @@ using Test
...
+        for combined in (false, true)
+            forces = forcecomponents(opsd, interactions, tm_idxs; combined)
+            w = optimize_weights(forces)
+            # Exact force balance is presumably impossible to achieve in this case.
+            # For a weaker test, check that weighted interactions need no additional weighting.
+            weighted_interactions = [key => w0*wi for ((key, w0), wi) in zip(interactions, w)]
+            wforces = forcecomponents(opsd, weighted_interactions, tm_idxs; combined)
+            @test all(f .* w' ≈ wf for (f, wf) in zip(forces, wforces))
+            wnew = optimize_weights(wforces)
+            @test_broken all(≈(1/length(interactions)), wnew)
```
Comment (timholy) [inline test/runtests.jl:292]: I'm puzzled about why this isn't passing. I worry there's some bug that is revealed by a complex example and isn't caught by the simple examples above.

Comment (timholy) [reply test/runtests.jl:292]: OK, it looks like it's simply a very, very flat basin:
image

**Comment by timholy** [comment]
In light of #20, let's rerun the tests.

**Pull request merged**